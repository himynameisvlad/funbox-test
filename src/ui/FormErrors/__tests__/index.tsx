import { shallow } from 'enzyme';
import * as React from 'react';
import renderer from 'react-test-renderer';

import { FormErrors } from '../index';

describe('FormErrors - ui component', () => {
  let wrapper;
  const formErrors = {
    errorOne: 'errorOne',
    errorTwo: 'errorTwo'
  };
  const FormErrorsComponent = <FormErrors formErrors={formErrors} />;

  beforeEach(() => {
    wrapper = shallow(FormErrorsComponent);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(FormErrorsComponent)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('doesn\'t render error if there is no one', () => {
    wrapper = shallow(<FormErrors formErrors={{}} />);

    expect(wrapper.find('p').length).toBe(0);
  });

  it('renders errors', () => {
    expect(wrapper.find('p').first().text()).toBe(formErrors.errorOne);
    expect(wrapper.find('p').at(1).text()).toBe(formErrors.errorTwo);
  });
})
