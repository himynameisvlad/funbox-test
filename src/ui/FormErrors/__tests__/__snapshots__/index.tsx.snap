// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormErrors - ui component renders correctly 1`] = `
<div
  className="fb-form-errors"
>
  <p>
    errorOne
  </p>
  <p>
    errorTwo
  </p>
</div>
`;
